11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/application_templates/MonoDeviceApplication.hpp"
+#include "nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+
+
+// TODO: Copy HelloSwapchain to set up a swapchain, draw to a surface. Or copy the new MPMC example.
+// TODO: Dynamically modify bloom size with mouse scroll/ +- keys
+//		 Notes: To do so must check required size against buffer size (expand buffer and recompile kernel if it starts being small), also after making kernel larger then small again
+//		        there's probably no need to shrink buffer and recompile kernel but would be nice to add that as well)
+// TODO: Refactor after FFT ext is back
+// TODO: Make sampling formats be #defined depending on how they were loaded on GPU side
+
+using namespace nbl;
+using namespace core;
+using namespace system;
+using namespace asset;
+using namespace video;
+
+
+#include "app_resources/common.hlsl"
+#include "nbl/builtin/hlsl/bit.hlsl"
+#include "utils.h"
+
+// In this application we'll cover buffer streaming, Buffer Device Address (BDA) and push constants 
+class FFTBloomApp final : public application_templates::MonoDeviceApplication, public application_templates::MonoAssetManagerAndBuiltinResourceApplication
+{
+	using device_base_t = application_templates::MonoDeviceApplication;
+	using asset_base_t = application_templates::MonoAssetManagerAndBuiltinResourceApplication;
+
+	// Compute Pipelines
+	smart_refctd_ptr<IGPUComputePipeline> m_firstAxisFFTPipeline;
+	smart_refctd_ptr<IGPUComputePipeline> m_lastAxisFFT_convolution_lastAxisIFFTPipeline;
+	smart_refctd_ptr<IGPUComputePipeline> m_firstAxisIFFTPipeline;
+
+	// Descriptor Sets
+	smart_refctd_ptr<IGPUDescriptorSet> m_firstAxisFFTDescriptorSet;
+	smart_refctd_ptr<IGPUDescriptorSet> m_lastAxisFFT_convolution_lastAxisIFFTDescriptorSet;
+	smart_refctd_ptr<IGPUDescriptorSet> m_firstAxisIFFTDescriptorSet;
```

Comment:
simplify code, use one descriptor set, and one layout, its fine to have extra "unused" descriptors

Reply:
also use texture arrays not arrays of textures